app/controllers/items_controller.rb

```diff
@@ -18,19 +18,29 @@ def update
     @merchant = Merchant.find(params[:merchant_id])
     @item = @merchant.items.find(params[:id])
   
-    if @item.update(item_params)
+
+    if params[:disable] == 'true'
+      @item.disabled!
+      flash[:success] = "Item successfully disabled."
+      redirect_to merchant_items_path(@merchant)
+    elsif params[:enable] == 'true'
+      @item.enabled!
+      flash[:success] = "Item successfully enabled."
+      redirect_to merchant_items_path(@merchant)
+    elsif @item.update(item_params)
       flash[:success] = "Item successfully updated."
+      redirect_to merchant_item_path(@merchant, @item)
     else
       flash[:error] = "Failed to update item."
```

Comment:
I'm wondering if this line can be tested?  Maybe it's just on my local machine but SimpleCov show that this line is not covered.  I'll double check the spec as well.